### PR TITLE
Sfl 164

### PR DESCRIPTION
**Pull request by saketPandhare-Webonise**

### CODE REVIEW

File: SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+protocol NibLoadableView: class { }
+ 
+extension NibLoadableView where Self: UIView {
+    
+    static var nibName: String {
+        // notice the new describing here
+        // now only one place to refactor if describing is removed in the future
+        return String(describing: self)
+    }
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift:17]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+extension UITableViewCell: NibLoadableView { }
+
+protocol ReusableView: class {}
+ 
+extension ReusableView where Self: UIView {
+    
+    static var reuseIdentifier: String {
+        return String(describing: self)
+    }
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift:34]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/Login/LoginViewController.swift [hunk 3]
```diff
@@ -48,6 +48,7 @@ class LoginViewController: UIViewController,LoginDelegate,UITextFieldDelegate, D
         initializeTextField()
         setSocialLogin()
         setAnalytics()
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/Login/LoginViewController.swift:51]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
+//
+//  SettingTableCell.swift
+//  SUNFLOWR
+
+import UIKit
+
+class SettingTableCell: UITableViewCell {
+
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift:9]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    
+    @IBOutlet weak var lblCellName: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift:18]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            
+        case SettingScreenCase.PrivacyPolicy.rawValue:
+            lblCellName.text = SettingScreenCellName.PrivacyPolicy
+            break
+            
+        default:
+            break
+        }
+    }
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift:57]: Please remove blank space

File: SUNFLOWR/SUNFLOWR/SettingsVC/SettingsVC.swift [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        super.viewDidLoad()
+
+        confNavigatonBar()
+        configureTableView()
+    }
+    
+    /// This function is used for Navigation bar configuration
+    func confNavigatonBar() {
+        setTitle(title: ScreenTitle.SETTINGS)
+        showLeftBarButtonWithImage(image: "back_arrow")
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/SettingsVC/SettingsVC.swift:23]: Can we take this image name from constants

File: SUNFLOWR/SUNFLOWR/Common Methods/WebServiceHandler.swift [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                    guard let data = response.data else {
+                        return
+                    }
+                    let errorResponse = self.nsdataToJSON(data: data as NSData)
+                    failure(errorResponse, statusCode)
+                }
+        }
+    }
+
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/Common Methods/WebServiceHandler.swift:70]: Please remove blank space

Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/Common Methods/WebServiceHandler.swift]: Done

File: SUNFLOWR/SUNFLOWR/SettingsVC/SettingsVC.swift (hunk 7 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/SettingsVC/SettingsVC.swift:23]: Not needed as image name is already in image asset ,the name of the same can be treated as constant

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift (hunk 6 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift]: Done

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift (hunk 5 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift]: Done

File: SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift (hunk 4 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/SettingsVC/TableCell/SettingTableCell.swift]: Done

File: SUNFLOWR/SUNFLOWR/Login/LoginViewController.swift (hunk 3 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/Login/LoginViewController.swift]: Done

File: SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift (hunk 2 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift]: Done

File: SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift (hunk 1 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/Extension/TableViewExtension.swift]: Done

**Pull request merged**